src/wp-includes/link-template.php

```diff
@@ -1618,14 +1618,17 @@ function get_edit_comment_link( $comment_id = 0, $context = 'display' ) {
 
 	$location = admin_url( $action ) . $comment->comment_ID;
 
+	// Ensure the $comment_id variable passed to the filter is always an ID.
+	$comment_id = (int) $comment->comment_ID;
+
 	/**
 	 * Filters the comment edit link.
 	 *
 	 * @since 6.7.0 The $comment_id and $context parameters are now being passed to the filter.
 	 *
 	 * @param string $location The edit link.
 	 * @param int    $comment_id Optional. Unique ID of the comment to generate an edit link.
-	 * @param int    $context    Optional. Context to include HTML entities in link. Default 'display'.
+	 * @param string $context    Optional. Context to include HTML entities in link. Default 'display'.
```

Comment:
This is an unrelated minor fix for an incorrect doc type that I noticed while fixing the main bug here.